src/main/java/com/ctc/wstx/sax/WstxSAXParser.java

```diff
@@ -1281,8 +1281,13 @@ public void dtdInternalEntityDecl(String name, String value)
     private void throwSaxException(Exception src)
         throws SAXException
     {
-        SAXParseException se = new SAXParseException(src.getMessage(), /*(Locator)*/ this, src);
-        ExceptionUtil.setInitCause(se, src);
+        SAXParseException se;
+        if (src instanceof SAXParseException) {
+            se = (SAXParseException) src;
+        } else {
+            se = new SAXParseException(src.getMessage(), /*(Locator)*/ this, src);
+            ExceptionUtil.setInitCause(se, src);
```

Comment:
This seemed unnecessary at first, until I realized that `SAXException` has its own 

    private Exception exception;

field separate from `cause` of JDK `Exception` (for legacy reasons, I assume).
So this calls is to update `cause`.